google/cloud/bigtable/internal/data_connection_impl_test.cc

```diff
@@ -1474,6 +1471,42 @@ TEST_F(DataConnectionTest, SampleRowsPermanentError) {
   EXPECT_THAT(samples, StatusIs(StatusCode::kPermissionDenied));
 }
 
+TEST_F(DataConnectionTest, SampleRowsBigtableCookie) {
+  auto mock = std::make_shared<MockBigtableStub>();
+  EXPECT_CALL(*mock, SampleRowKeys)
+      .WillOnce(
+          [this](auto context, auto const&, v2::SampleRowKeysRequest const&) {
+            metadata_fixture_.SetServerMetadata(
+                *context, {{}, {{"x-goog-cbt-cookie-routing", "routing"}}});
+            auto stream = std::make_unique<MockSampleRowKeysStream>();
+            EXPECT_CALL(*stream, Read).WillOnce(Return(TransientError()));
+            return stream;
+          })
+      .WillOnce(
+          [this](auto context, auto const&, v2::SampleRowKeysRequest const&) {
+            auto headers = metadata_fixture_.GetMetadata(*context);
+            EXPECT_THAT(headers,
+                        Contains(Pair("x-goog-cbt-cookie-routing", "routing")));
+            auto stream = std::make_unique<MockSampleRowKeysStream>();
+            EXPECT_CALL(*stream, Read).WillOnce(Return(PermanentError()));
+            return stream;
+          });
+
+  auto mock_b = std::make_unique<MockBackoffPolicy>();
```

Comment:
rename policy_mock?

Reply:
No thanks. The name is consistent with other tests in the file.